### PR TITLE
[C#] dictionary ex. tests with only one assert

### LINKED ISSUE #1557: [C#] Follow up unresolved review points on PR 1497 dictionaries
Follow up on [PR 1497](https://github.com/exercism/v3/pull/1497) including eliminating multiple asserts in test.

### PR DESCRIPTION
**Pull request by valentin-p**
Closes #1557

### CODE REVIEW

**Review by mikedamay — COMMENTED** [review]
This looks good to me.

My only quibble is that we lose a lot of information in the assert messages.  Single assert is @ErikSchierboom / @iHiD 's thing.  It is to do with test runner output (which I hope to see tomorrow).  I wonder if it would be better to compare arrays of actual and expected values.

File: languages/csharp/exercises/concept/dictionaries/DictionariesTests.cs [hunk 1]
```diff
@@ -11,31 +11,31 @@ public void Empty_dictionary()
     [Fact(Skip = "Remove this Skip property to run this test")]
     public void Existing_dictionary()
```
Comment (valentin-p) [inline languages/csharp/exercises/concept/dictionaries/DictionariesTests.cs:12]: Would that be easy enough for the student to understand at this exercise?
```
    public void Existing_dictionary()
    {
        var prePopulated = DialingCodes.GetExistingDictionary();
        var list = prePopulated.SelectMany(x => new[] { x.Key.ToString(), x.Value }.Select(y => y));
        var expected = new List<string> { "1", "United States of America", "55", "Brazil", "91", "India" };
        Assert.Equal(expected, list);
    }
```

**Comment by ErikSchierboom** [comment]
I agree that while the current approach solves the issue, maybe there is another way to solve this. Consider the following test:

```csharp
[Fact(Skip = "Remove this Skip property to run this test")]
public void Add_country_to_empty_dictionary()
{
    var idcd = DialingCodes.AddCountryToEmptyDictionary(44, "United Kingdom");
    Assert.Equal(1, idcd.Count);
    Assert.Equal("United Kingdom", idcd[44]);
}
```

This test could be split up into two tests:

```csharp
[Fact(Skip = "Remove this Skip property to run this test")]
public void Add_country_to_empty_dictionary_has_count_of_one()
{
    var idcd = DialingCodes.AddCountryToEmptyDictionary(44, "United Kingdom");
    Assert.Equal(1, idcd.Count);
}

[Fact(Skip = "Remove this Skip property to run this test")]
public void Add_country_to_empty_dictionary_adds_country_at_correct_index()
{
    var idcd = DialingCodes.AddCountryToEmptyDictionary(44, "United Kingdom");
    Assert.Equal("United Kingdom", idcd[44]);
}
```

**Comment by mikedamay** [comment]
@ErikSchierboom your solution works for `Add_country_to_empty_dictionary`.  Do you think @valentin-p / my array-based approach works for `Add_country_to_existing_dictionary` and `Update_country_name_in_dictionary`?

**Comment by mikedamay** [comment]
@valentin-p You are using `SelectMany()` in the tests.  The track policy is to avoid concepts that are not depeendencies even in test code if possible.  LINQ is a way off at present.  A nuissance but easily worked around.

**Comment by ErikSchierboom** [comment]
>  Do you think @valentin-p / my array-based approach works for Add_country_to_existing_dictionary and Update_country_name_in_dictionary?

Whic approach is that exactly?

**Review by ErikSchierboom — APPROVED** [review]
Great!

**Comment by ErikSchierboom** [comment]
@mikedamay You okay with merging this?

**Review by mikedamay — APPROVED** [review]
Yep.  Let's merge this.

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**